### PR TITLE
feat: value normalizer

### PR DESCRIPTION
**Pull request by andreapiras**
☝ This title will appear in the changelog - keep it meaningful and follow [the commit lint format](https://github.com/transferwise/neptune-web/blob/master/CONTRIBUTING.md#versioning-and-commit-lint).

## 🖼 Context

Our components return values in different format. This wrapper aims to normalized the different return and make it consistent and predictable

## ✅ Checklist

- [ ] Changes are tested and all tests pass
- [ ] Changes meet [accessibility standards](https://github.com/transferwise/neptune-web/blob/master/ACCESSIBILITY.md) and there are no violations in the console
- [ ] Changes work in all supported browsers (don't forget IE11)
- [ ] You've updated the documentation if necessary

### CODE REVIEW

**Comment by StevePole** [comment]
Feels like we're working around other problems, rather than addressing the core issue, which is that our components return inconsistent things.

What about if we changed all component to return a native compliant event with a target and value?  Or alternatively what if we implemented a React `<Input ... />` that broadcast just the value?

**Comment by StevePole** [comment]
With the type validators, I wouldn't implement isNumeric or isEvent in this file as these aren't JS types.  There could possibly be an isClass, or isInstanceOf (that has two params).

Keeping inline with JSON schema we might consider 'format validators' that check, for example, if a string is: numeric, an iso date, or a base64url.

File: packages/dynamic-flows/src/WithNormalizer/WithNormalizer.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+const WithNormalizer = ({ children }) => {
+  const handleOnChange = (event) => {
+    let value = event;
+
+    if (isEvent(event)) {
+      value = isNumeric(event.target.value) ? parseFloat(event.target.value) : event.target.value;
+    } else if (event && (event.value || isNumber(event.value))) {
+      value = event.value;
+    } else if (event.status) {
+      value = event.status;
```
Comment (dozortsev) [inline packages/dynamic-flows/src/WithNormalizer/WithNormalizer.js:14]: why `event` would have `value`/`status` props?

Comment (dozortsev) [reply packages/dynamic-flows/src/WithNormalizer/WithNormalizer.js:14]: it feels like to start use it everywhere we will need grow complexity of these checks, e.g tomorrow you would need add check for `data` prop and then for `initialValue` etc etc.

**Comment by andreapiras** [comment]
> We better refactor our components to consistent return value, instead of fixing "mess" with wrapper/abstraction.
> 
> I haven't done much thought about this and didn't do any research but atm I'm thinking in favor of [native Event object](https://developer.mozilla.org/en-US/docs/Web/API/Event) as it's data could be useful for consumers in their business logic / UI.

This is on our roadmap, we'll centralize this function first and then update our component's API to emit consistently events

File: packages/dynamic-flows/src/WithNormalizer/validation/event-utilities/index.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+const normalizeEvent = (event, type) => {
+  let value = event;
+
+  if (event) {
+    if (isSyntheticEvent(event)) {
+      value = type === 'number' ? parseFloat(event.target.value) : event.target.value;
+    } else if (isNumber(event.value) || event.value) {
+      // Checkbox,Select
+      value = event.value;
```
Comment (dozortsev) [inline packages/dynamic-flows/src/WithNormalizer/validation/event-utilities/index.js:20]: what about `event.status`?

Comment (andreapiras) [reply packages/dynamic-flows/src/WithNormalizer/validation/event-utilities/index.js:20]: There was no handling in the previous logic so we decided to remove it. We can add it in a second iteration if needed

Comment (dozortsev) [reply packages/dynamic-flows/src/WithNormalizer/validation/event-utilities/index.js:20]: 👌

**Pull request merged**